specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/CommonDefinitions.json

```diff
@@ -0,0 +1,2246 @@
+          "x-ms-enum": {
+            "name": "AccessMode",
+            "modelAsString": true
+          }
+        },
+        "shareName": {
+          "description": "NFS Azure file share name.",
+          "type": "string"
+        }
+      }
+    },
+    "SmbStorage": {
+      "description": "SMB storage properties",
+      "type": "object",
+      "properties": {
+        "host": {
+          "description": "The host name or IP address of the SMB server.",
+          "type": "string"
+        },
+        "shareName": {
+          "description": "The path to the SMB shared folder.",
+          "type": "string"
+        },
+        "username": {
+          "description": "The user to log on to the SMB server.",
+          "type": "string"
+        },
+        "domain": {
+          "description": "The domain name for the user.",
+          "type": "string"
+        },
+        "password": {
+          "description": "The password for the user.",
+          "type": "string",
+          "x-ms-secret": true
+        },
+        "accessMode": {
+          "description": "Access mode for storage",
+          "enum": [
+            "ReadOnly",
+            "ReadWrite"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "AccessMode",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "CertificateKeyVaultProperties": {
+      "description": "Properties for a certificate stored in a Key Vault.",
+      "type": "object",
+      "properties": {
+        "identity": {
+          "description": "Resource ID of a managed identity to authenticate with Azure Key Vault, or System to use a system-assigned identity.",
+          "type": "string"
+        },
+        "keyVaultUrl": {
+          "description": "URL pointing to the Azure Key Vault secret that holds the certificate.",
+          "type": "string"
+        }
+      }
+    },
+    "Certificate": {
+      "description": "Certificate used for Custom Domain bindings of Container Apps in a Managed Environment",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Certificate resource specific properties",
+          "type": "object",
+          "properties": {
+            "provisioningState": {
+              "description": "Provisioning state of the certificate.",
+              "enum": [
+                "Succeeded",
+                "Failed",
+                "Canceled",
+                "DeleteFailed",
+                "Pending"
+              ],
+              "type": "string",
+              "readOnly": true,
+              "x-ms-enum": {
+                "name": "CertificateProvisioningState",
+                "modelAsString": true
+              }
+            },
+            "deploymentErrors": {
```

Comment:
>deploymentErrors [](http://example.com/codeflow?start=13&length=16)

will this be a string with a concatenated list of errors? or should this be converted in an array?

Reply:
No. It is error string as you can see in example files in https://github.com/Azure/azure-rest-api-specs/pull/29822.

IN addition, this change is added in https://github.com/Azure/azure-rest-api-specs/pull/29822 which already get arm signed off.